command/src/scm_socket.rs

```diff
@@ -23,30 +23,32 @@ pub struct ScmSocket {
 }
 
 impl ScmSocket {
-    pub fn new(fd: RawFd) -> ScmSocket {
+    pub fn new(fd: RawFd) -> anyhow::Result<Self> {
         unsafe {
             let stream = net::UnixStream::from_raw_fd(fd);
-            let _ = stream.set_nonblocking(false).map_err(|e| {
-                error!("could not change blocking status for stream: {:?}", e);
-            });
+            let _ = stream
+                .set_nonblocking(false)
```

Comment:
We should investigate on why it could fail

Reply:
there is very little chance it could fail, so we might as well just propagate the error for an easier debugging in this very unlikely event